### ISSUE TITLE
Allow timber_post_get_meta_pre filter to short circuit or alter Timber\Post::get_post_custom()

### ISSUE THREAD

**Issue by aj-adl** [timber/timber#2011]
## Is your feature request related to a problem? Please describe.

When dealing with large collections of items, sometimes the eager loading behaviour of Timber works against us and makes for heavy arrays that consume a lot of memory. We notice this particularly when the items we're querying for / working with have a lot of ACF flex content, as this really bulks out the meta.

It looks like there has already been an attempt to solve this in the get_post_custom() function, but the filter that runs has no opportunity to affect the behaviour beyond the callback, the `get_post_custom()` will be called anyway, all the time.

This is the "first" issue, and one that's much easier to deal with, then the "real" issue that I've actually had multiple times, which is the same thing on Terms, where the meta fetching and capabilities are very intertwined with ACF, rather than using the now widely available term_meta functions and using ACF if necessary.

Pull request for both inbound, this one is easy, and sets the pattern/tone for the next harder one related to terms.

## Describe the solution you’d like

Some rough pseudo code below
```php
/* Prevent eager loading of Post Meta */
add_filter( timber_post_get_meta_pre, function() { return false; } );

/* only eager load certain meta keys */

add_filter( 'timber_post_get_meta_pre', function($customs, $pid, $timberPost ) {
    $keys = [ 'key1', 'key2', 'key3' ];
    $customs = [];
    foreach( $keys as key ){
        $customs[ $key] = get_post_meta( $pid, $key );
    }
    return $customs;
}, 10 , 3 );
```
## Describe alternatives you’ve considered

It's possible to filter the values returned to the import statment, but that's only after the data is fetched from the db, so this way aims to reduce the overhead of that on big queries / collections.

## Additional context

unsure if should be implemented as `timber_post_get_meta_pre` or `timber/post/meta/pre` ?

**Comment by gchtr** [comment]
Hey Alex!

There are quite a few changes coming for fetching meta values in 2.x. For `timber_post_get_meta_pre`, there’s a new filter in [`get_meta_values()`](https://github.com/timber/timber/blob/ccefcb5cfa1abe2faaeac533104e229552c79ceb/lib/Post.php#L574) called `timber/post/pre_get_meta_values`:

https://github.com/timber/timber/blob/ccefcb5cfa1abe2faaeac533104e229552c79ceb/lib/Post.php#L577-L607

But as I see, you’ll only want certain values to be excluded. We didn’t account for that yet and I think what you have is a good use case. We definitely care about performance and anything that makes it better is very welcome. So I’m eager to look at your pull requests 👍.

**Comment by aj-adl** [comment]
Thanks Lucas, PR is submitted, I totally forgot the check out what's in the 2.0 branch on this, as I mentioned this is more of an easy 'part 1' to the bigger 'part 2' which is improving term meta loading and eager-loading.

**Comment by jarednova** [comment]
Closed by #2014

**Comment by gchtr** [comment]
@aj-adl Can we expect the bigger «part 2» for the term meta loading and eager-loading, now that this pull request is merged? I’m curious for what you have in mind. 😉

**Comment by aj-adl** [comment]
@gchtr I've been catching up on 2.0 and it seems that it's been fixed in #2019  with `Term::get_meta_values` now mirroring the same functionality and using WordPress's built in term meta functions.

Our original issue was that in the 1.9.x branches, getting term meta relied on ACF (as setting term meta used to rely on ACF!), which meant that if we attached a flex content field or related terms / objects to a term then these would all get eager loaded by the ACF integration..

I could look at a PR back porting something, but honestly I'd rather help out with other 2.x things so it can be released sooner!

**Comment by jarednova** [comment]
@aj-adl — I'd love to talk more about exactly that. Can you email me at `jared@upstatement.com`? I have some ideas ...
